packages/container/libraries/strongly-typed/src/container.spec.ts

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable jest/expect-expect */
+import { beforeEach, describe, expect, it } from '@jest/globals';
+
+import 'reflect-metadata';
+
+import { Container, injectable } from 'inversify';
+
+import { TypedContainer } from './container';
+
+describe('interfaces', () => {
+  @injectable()
+  class Foo {
+    public foo: string = '';
+  }
+
+  @injectable()
+  class Bar {
+    public bar: string = '';
+  }
+
+  describe('Container', () => {
+    let foo: Foo;
+    let foos: Foo[];
+
+    beforeEach(() => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      foo;
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      foos;
+    });
+
+    describe('no binding map', () => {
+      let container: TypedContainer;
+
+      beforeEach(() => {
+        container = new Container() as TypedContainer;
```

Comment:
This type assertion is currently **required**. That is, you **can't** just write:

```ts
let container: TypedContainer;
container = new Container();
```

...because TypeScript considers the two types incompatible. This makes me feel a little uneasy, but I think it's unavoidable without touching upstream types.

Reply:
As discussed, we'll give consumers a couple of options to use this library:

```ts
container = new Container() as TypedContainer<Mapping>;
container = wrapContainer<Mapping>(new Container());
container = new TypedContainer<Mapping>();
```

Reply:
Actually @notaphplover I've thought about it some more, and I *don't* think we should have the wrapper method (in the interests of keeping the API footprint small).

I think it offers no benefits over the other two methods, and instead pulls in the worst things about them. It's just performing a type assertion under the hood, but this is now less transparent. Since the function requires a non-`type` import, it also won't tree shake this dependency out of the final build.

Reply:
Let's forget about the wraper then